subsys/logging/log_core.c

```diff
@@ -352,6 +359,43 @@ void log_n(const char *str,
 	}
 }
 
+int log_format_set(char *backend_name, uint32_t log_type)
+{
+	const struct log_backend *backend = log_backend_get_by_name(backend_name);
+
+	if (log_type >= ARRAY_SIZE(log_format_table)) {
+		return -EINVAL;
+	}
+
+	if (backend != NULL) {
+		if (backend->api->format_set == NULL) {
+			return -ENOTSUP;
+		} else {
+			return backend->api->format_set(backend, log_type);
+		}
+	} else {
+		return -EINVAL;
+	}
+}
+
+const struct log_backend *log_format_set_all_backends(uint32_t log_type)
+{
+	const struct log_backend *backend;
+	const struct log_backend *failedBackend = NULL;
+
+	for (int i = 0; i < log_backend_count_get(); i++) {
+		backend = log_backend_get(i);
+		if (log_backend_is_active(backend)) {
```

Comment:
Two items.
1. Not sure why we are saying that the backend needs to be active.
2. If it is truly necessary, then the API name should reflect that.

Reply:
Changed the name of the API to reflect the setting of log format for all active backends.

Reply:
> Unless I have missed it, the work appears to be incomplete. For example, I think the routine log_output_msg2_process() still has the following check. if (IS_ENABLED(CONFIG_LOG_MIPI_SYST_ENABLE) && flags & LOG_OUTPUT_FLAG_FORMAT_SYST) { log_output_msg2_syst_process(output, msg, flags); return; } There may be other locations that need to be cleaned up as well.

@peter-mitsis Have not modified the log_output_msg2_syst_process routine for now , since other backends except UART are still using the old implementation. Wanted to seek some reviews/approvals on the design part and following that I will make changes to log_output_msg2_syst_process routine along with other backends.